net/srt/Portfile

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+github.setup        Haivision srt 1.5.2 v
+revision            0
+
+categories          net multimedia comms
+license             MPL-2
+maintainers         @jasonliu-- openmaintainer
+
+description         transport protocol for ultra-low latency live \
+                    video and audio streaming
+long_description    Secure Reliable Transport (SRT) is a \
+                    ${description}, as well as for generic bulk data \
+                    transfer. SRT provides reliable transmission \
+                    similar to TCP\; however, it does so at the \
+                    application layer, and uses UDP as the underlying \
+                    transport layer. This package contains the \
+                    open-source reference implementation of the SRT \
+                    library, and was written by Haivision, the \
+                    original developers of the SRT protocol.
+
+checksums           rmd160  e87b64346e59ff58c4cb1020ebd79733f7e53fdd \
+                    sha256  3bcde30be184aa9b3a7d4a13e1a72f89ac1d221a9aaa9860bdf2c7b81fd11ae0 \
+                    size    1699992
+
+compiler.cxx_standard   2011
+
+configure.args-append   -DENABLE_APPS=OFF \
+                        -DENABLE_LOGGING=OFF \
+                        -DENABLE_ENCRYPTION=OFF \
+                        -DENABLE_SHOW_PROJECT_CONFIG=ON
+
+pre-test {
+    if {![variant_isset tests]} {
+        ui_error "'tests' variant must be activated to enable test support"
+        error "Please enable the 'tests' variant and try again"
+    }
+}
+
+variant utils description {Build command-line tools} {
+    configure.args-delete   -DENABLE_APPS=OFF
+}
+
+variant logging description {Enable logging} {
+    configure.args-delete   -DENABLE_LOGGING=OFF
+}
+
+variant heavy_logging requires logging \
+    description {Enable heavy logging} \
+{
+    configure.args-append   -DENABLE_HEAVY_LOGGING=ON
+}
```

Comment:
do we need this three variants or can/should it just be enabled by default. The use of variants is typically only done when optional things require "heavy" dependencies or when there are functional options as you do here with openssl/alternatives and the "test". The three above only change configure arguments and IMHO should just be enabled by default as there is no (obvious) drawback for end-users.

Reply:
The first three variants allow users to build slimmed-down versions of the port if they want. Yes, you're correct that variants are used when optional things require many additional dependencies, but variants can also be used to expose the various compile options which may be useful to users. Some ports, particularly ones dealing with graphics or audio/video libraries, come with many, many compile options in the source code. Users who are on older hardware or may not want ports to take up so much of their hard drive space may want to build ports with certain compile options turned off.

Reply:
well... the problem is that none of the (non-default) variants get build by the CI or buildbots so it's not ideal in that sense to make sure that things actually work. 
There are pretty much always many configure options available to packages, it's up to the maintainer to pick a reasonable set of things and try to avoid a wild-west of variants. How much disk-space saving are we talking about here - I kind of doubt that this would be a sufficient amount to "defend" the need of variants, but without any data that's hard to tell

Reply:
This library is part of a wider effort that I'm working on, to try to add OBS Studio to MacPorts. In this particular case for srt, these variants help with troubleshooting and development work, without needing to create a separate `-devel` subport. With so many moving parts to OBS, the variants have helped me to try and track down linking and runtime issues. From my experience, the vast majority of MacPorts users don't mess around with variants anyway. Usually it's people (people who I support in my university department IT office, so again my own personal experience) who like to use MacPorts packages (typically libraries) as a development platform.

Reply:
@reneeotten That seems like a good reason for him to keep all the variants.